nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/cluster/ZooKeeperClientConfig.java

```diff
@@ -70,7 +75,10 @@ public static ZooKeeperClientConfig createConfig(final Properties properties) {
         if (connectString == null || connectString.trim().isEmpty()) {
             throw new IllegalStateException("The '" + NiFiProperties.ZOOKEEPER_CONNECT_STRING + "' property is not set in nifi.properties");
         }
-
+        final String cleanedConnectString = cleanConnectString(connectString);
+        if (cleanedConnectString.isEmpty()) {
+            throw new IllegalStateException("The '" + NiFiProperties.ZOOKEEPER_CONNECT_STRING + "' property is set but contains unrecognized data in nifi.properties");
```

Comment:
This message could probably be made a little more clear - what is "unrecognized data"? Perhaps spell out that the expected format is a comma-separated list of <hostname>:<port> pairs.